coderd/database/pubsub/pubsub.go

```diff
@@ -352,11 +353,61 @@ func (p *PGPubsub) recordReconnect() {
 	}
 }
 
+// logDialer is a pq.Dialer and pq.DialerContext that logs when it starts
+// connecting and when the TCP connection is established.
+type logDialer struct {
+	logCtx context.Context
+	logger slog.Logger
+	d      net.Dialer
+}
+
+var _ pq.Dialer = logDialer{}
+var _ pq.DialerContext = logDialer{}
+
+func (d logDialer) Dial(network, address string) (net.Conn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	return d.DialContext(ctx, network, address)
+}
+
+func (d logDialer) DialTimeout(network, address string, timeout time.Duration) (net.Conn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	return d.DialContext(ctx, network, address)
+}
+
+func (d logDialer) DialContext(ctx context.Context, network, address string) (net.Conn, error) {
+	deadline, hasDeadline := ctx.Deadline()
+	timeoutMS := 0
+	if hasDeadline {
+		timeoutMS = int(time.Until(deadline) / time.Millisecond)
+	}
+
+	d.logger.Info(d.logCtx, "pubsub dialing postgres", slog.F("network", network), slog.F("address", address), slog.F("timeout_ms", timeoutMS))
```

Comment:
if you do something like

```
logger := d.logger.With(slog.F("network", network), ...)
```

for these fields, then you only have to set them once.